### ISSUE TITLE
Ethos 1.6.0 and GPS coordinates Longitude and Latitude are incorrect -- 10 times to high

### ISSUE THREAD

**Issue by Storchy1** [FrSkyRC/ETHOS-Feedback-Community#4941]
the value of the GPS coord Lan and Lat are 10 times to high since Ethos 1.6.0
Lat = system.getSource({name="GPS", options=OPTION_LATITUDE}:value()
normal the float value Lat in germany is around 49.1xxxxx
in Ethos 1.6.0 the value is now 491,.......
the same for Lon =  system.getSource({name="GPS", options=OPTION_LONGITUDE}:value()
the value shoud be 11.3xxxx and is 113.xxxx
Please divide the value by 10 
thanks for your great work

**Comment by flyingbandit** [comment]
Same here

**Comment by frskyrc-ben** [comment]
I don't know if it's related or not, but users are also reporting the GPS Widget is now broken and they cannot enter the coordinates accordingly.

> can't input coordinates higher than 9° N and 18° W 

![image](https://github.com/user-attachments/assets/95b76caf-3111-4177-ae5e-81eac41952b7)

**Comment by flyingbandit** [comment]
> I don't know if it's related or not, but users are also reporting the GPS Widget is now broken and they cannot enter the coordinates accordingly.
> 
> > can't input coordinates higher than 9° N and 18° W
> 
> ![image](https://private-user-images.githubusercontent.com/108486338/399393907-95b76caf-3111-4177-ae5e-81eac41952b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU2NzgxNTMsIm5iZiI6MTczNTY3Nzg1MywicGF0aCI6Ii8xMDg0ODYzMzgvMzk5MzkzOTA3LTk1Yjc2Y2FmLTMxMTEtNDE3Ny1hZTVlLTgxZWFjNDE5NTJiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIzMVQyMDQ0MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNTRhMDNkMjBkYTkxNGIwNmI2OTQ5ZDI5MWE4YTk5MGQzMTQ4MmJiY2FkYTZhZjM5OWUwZWNmZjg4MWRjODMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4YIQZplswcPdH9F3UgEOFbn3J7CaUAGNewhy5Tc8G3g)

Yes this was the same for me. I wasted a lot of time messing around with different gps widgets and then got it downgraded to 1.5.19 which fixed the GPS cords so I could carry out using the ethos yaapu mapping widget - it’s a really decent widget if you want an alternative to the GPS widget that in built in. You can convert huge areas of a map.

**Comment by msromike** [comment]
I think it is too much of a coincidence that the values being reported are ten times too high, and the input fields in the widget are now constrained to 10% of the correct range.  They seem related.

**Comment by msromike** [comment]
> I don't know if it's related or not, but users are also reporting the GPS Widget is now broken and they cannot enter the coordinates accordingly.
> 
> > can't input coordinates higher than 9° N and 18° W
> 

It is too much of a coincidence that the values being reported are ten times too high, and the input fields in the widget are now constrained to 1/10th of the correct range. They seem related.

**Comment by itizmeunderscoreca** [comment]
> I don't know if it's related or not, but users are also reporting the GPS Widget is now broken and they cannot enter the coordinates accordingly.
> 
> > can't input coordinates higher than 9° N and 18° W
> 
Same issue here

**Comment by msromike** [comment]
> > I don't know if it's related or not, but users are also reporting the GPS Widget is now broken and they cannot enter the coordinates accordingly.
> > > can't input coordinates higher than 9° N and 18° W
> 
> Same issue here

I bought my Geolink 2 partially for this widget, and.... :(

**Comment by Storchy1** [comment]
I also tested with my Geolink & Geolink2 from Spirit. The same Problems the values are 10 times to high

**Comment by bsongis-frsky** [comment]
The problem of the GPS Widget is reported in #4908 and fixed already.
Here is the same problem in Lua scripts.

**Comment by Storchy1** [comment]
Thanks a lot of your work!
Closed!

**Comment by bsongis-frsky** [comment]
One second. I need to check everywhere a GPS value is displayed / used before closing the issue.

**Comment by Storchy1** [comment]
Sorry i do not know this

**Comment by bsongis** [comment]
Ethos has a punition system:
For each issue found we have to introduce a new non regression test to ensure the bug won't happen again.

**Comment by bsongis-frsky** [comment]
While writing the automatic tests I see that the default Latitude / Longitude of the Map Widget are wrong too!

![image](https://github.com/user-attachments/assets/b21445de-8b85-4c0d-90f2-cc2d1090be5f)

**Comment by msromike** [comment]
The defaults are unintuitive, But, Longitude is labeled using North and South when they should be East and West.  Also, the individual values cannot be edited for the full range of Lattitude from 90 deg North to 90 deg South, nor the full range of Longitude from 180 deg East to 180 deg West.

**Comment by bsongis-frsky** [comment]
True as well!

**Comment by bsongis-frsky** [comment]
This is the result of the new non regression test:

![image](https://github.com/user-attachments/assets/10874187-e5d8-47b9-b411-e99ad1dc1c78)

**Comment by bsongis-frsky** [comment]
It's not finished. It seems to me that the point is not at the right position!

![image](https://github.com/user-attachments/assets/eb634418-c131-4dc9-b076-33d880955a5b)

**Comment by msromike** [comment]
I think the convention is East first then West while you are in there fixing stuff.

**Comment by bsongis-frsky** [comment]
What?

**Comment by msromike** [comment]
Well you have north and south defaults fine, but usually you see it as east and then west, not west and then east on your longitude fields, in context of your defaults.  Not a big deal, and i don't know why its east before west, but pretty sure it is that way in most apps and devices.

**Comment by msromike** [comment]
Maybe because the sun comes up in the east, LOL.  For instance, one most likely would say "from the North to the South and from the East to the West," in a conversation.

**Comment by bsongis-frsky** [comment]
Better?

![image](https://github.com/user-attachments/assets/c27e48dd-e74a-44cc-96dd-d26828bb5885)

**Comment by bsongis-frsky** [comment]
GPS calculations correct, and point at the right position!

![image](https://github.com/user-attachments/assets/31957e34-0310-4dce-ad6c-2101de84ae87)

**Comment by msromike** [comment]
Stellar!  Can't wait for the release.

**Comment by bsongis-frsky** [comment]
More non regression tests added with this screen:

![image](https://github.com/user-attachments/assets/34d3af84-ace8-4d9b-95a5-5c2fe5fb96a3)

- GPS map
- GPS position displayed in a Value widget
- GPS position displayed in a Lua widget `textValue` and `value`
- Distance between 2 points calculations checked

**Comment by msromike** [comment]
Nice.  Would you consider sharing your LUA widget?  I would like to look at it to use as a template.

**Comment by bsongis** [comment]
The one in the 2nd column, 2nd row? It is just a widget for my automatic tests. I can share it but it is there just for my automatic tests. With 0 control. The idea is just to avoid any further bug in this area ...

```lua
-- Lua GPS test

local function paint(widget)
    local w, h = lcd.getWindowSize()
    local sensor = system.getSource("GPS")
    lcd.color(ORANGE)
    lcd.drawText(w/2, h/2 - 20, sensor:stringValue(), TEXT_CENTERED)
    lcd.color(YELLOW)
    lcd.drawText(0, h/2 + 10, sensor:value({options=OPTION_LATITUDE}), TEXT_LEFT)
    lcd.drawText(w, h/2 + 10, sensor:value({options=OPTION_LONGITUDE}), TEXT_RIGHT)
end

local function init()
    system.registerWidget({key="gps", name="GPS", paint=paint})
end

return {init=init}
```

**Comment by bsongis** [comment]
It has no wakeup, no check if the sensor exists, no check if the sensor has a valid value ... so definitely not a good example. You should rather have a look in this repository, there is an examples directory which is better!

**Comment by msromike** [comment]
I can make that work, thanks!  Will add the error checking. just needed the skeleton.

**Comment by bsongis-frsky** [comment]
Interesting link with #4765  as it seems that the symbol used as a decimal separator in my screenshot above is the comma, when it should be a punct!

**Comment by Solarjojo** [comment]
> This is the result of the new non regression test:
> 
> ![image](https://github.com/user-attachments/assets/10874187-e5d8-47b9-b411-e99ad1dc1c78)


I Think , the most people get the maps from "Hobby4Live" and then transfer ist in the same way in the configurationscreen.

Leftcorner in the Top of the map  2 Values,

Right corner at the bottom of the Map. 2 Values

This is the way to avoid frustration of the most. easy logic.

**Comment by bsongis** [comment]
Good idea. But too late for 1.6.1 as it means new translations for "Left top corner" and "Right bottom corner"

**Comment by Solarjojo** [comment]
Maybe, for the next stable Version.
😎
